src/loki/worker.cc

```diff
@@ -262,6 +267,43 @@ void loki_worker_t::set_interrupt(const std::function<void()>* interrupt_functio
   reader->SetInterrupt(interrupt);
 }
 
+// Check if total arc distance exceeds the max distance limit for disable_hierarchy_pruning.
+// If true, add a warning and set the disable_hierarchy_pruning costing option to false.
+void loki_worker_t::check_hierarchy_distance(Api& request, bool is_matrix) {
```

Comment:
I'd rename this to `pair_wise` or so to decouple from the action

Reply:
Done. Good suggestion.